homeassistant/components/stt/__init__.py

```diff
@@ -72,9 +72,18 @@
 @callback
 def async_default_engine(hass: HomeAssistant) -> str | None:
     """Return the domain or entity id of the default engine."""
-    return next(
-        iter(hass.states.async_entity_ids(DOMAIN)), None
-    ) or async_default_provider(hass)
+    component: EntityComponent[SpeechToTextEntity] = hass.data[DOMAIN]
+
+    default_entity_id: str | None = None
+
+    for entity in component.entities:
+        if entity.platform and entity.platform.platform_name == "cloud":
```

Comment:
stt integration should not be aware of other integrations.  If we need to have this preference we should add a function that other integrations can use to register preferred engines.

Reply:
We should remove `stt.async_default_engine` and `tts.async_default_engine` in an upcoming release. I think it's OK to have the hard coded reference to "cloud" here until then?

This or a separate PR could update the docstrings to mention the functions are deprecated and will be removed very soon.

Reply:
Ok, good.